File: 2000_reverse_prefix_of_word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        first_occurence = 0
        for i in range(len(word)):
            if word[i] == ch:
                first_occurence = i
                break

        if first_occurence:
            return word[:first_occurence + 1][::-1] + word[first_occurence + 1:]
        return word

File: JavaScript/2000_reverse_prefix_of_word.js
/**
 * @param {string} word
 * @param {character} ch
 * @return {string}
 */
var reversePrefix = function(word, ch) {
    const firstIndex = word.indexOf(ch);
    const reverse = word.slice(0, firstIndex + 1).split('').reverse().join('');
    return reverse + word.slice(firstIndex + 1);
};

